src/containers/team-leaderboard/TeamLeaderboard.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import LeaderboardTabs, {
+  TabInfo,
+} from '../../components/leaderboard/leaderboard-tabs/LeaderboardTabs';
+import { C4CState } from '../../store';
+import PageHeader from '../../components/pageheader/PageHeader';
+import styled from 'styled-components';
+import {
+  TeamLeaderboardWeeklyReducerState,
+  TeamLeaderboardMonthlyReducerState,
+  TeamLeaderboardYearlyReducerState,
+  TeamLeaderboardAllTimeReducerState,
+} from './ducks/types';
+import { connect, useDispatch } from 'react-redux';
+import { AsyncRequestKinds } from '../../utils/asyncRequest';
+import { getTeamsLeaderboard } from './ducks/thunks';
+import {
+  LeaderboardPreviousDays,
+} from '../../components/leaderboard/ducks/types';
+import { leaderboardItemsToTabItems } from '../../components/leaderboard/leaderboard-tab/LeaderboardTab';
+
+const LeaderboardContentContainer = styled.div`
+  padding: 100px 134px;
+`;
+
+const LeaderboardContainer = styled.div`
+  margin: 80px 0px 40px;
+`;
+
+interface TeamLeaderboardProps {
+  readonly teamLeaderboardItemsWeekly: TeamLeaderboardWeeklyReducerState['teamLeaderboardWeekly'];
+  readonly teamLeaderboardItemsMonthly: TeamLeaderboardMonthlyReducerState['teamLeaderboardMonthly'];
+  readonly teamLeaderboardItemsYearly: TeamLeaderboardYearlyReducerState['teamLeaderboardYearly'];
+  readonly teamLeaderboardItemsAllTime: TeamLeaderboardAllTimeReducerState['teamLeaderboardAllTime'];
+}
+
+const TeamLeaderboard: React.FC<TeamLeaderboardProps> = ({
+  teamLeaderboardItemsWeekly,
+  teamLeaderboardItemsMonthly,
+  teamLeaderboardItemsYearly,
+  teamLeaderboardItemsAllTime,
+}) => {
+  const dispatch = useDispatch();
+
+  const [weeklyTab, setWeeklyTab] = useState<TabInfo>({
+    name: 'Weekly',
+    content: [],
+  });
+  const [monthlyTab, setMonthlyTab] = useState<TabInfo>({
+    name: 'Monthly',
+    content: [],
+  });
+  const [yearlyTab, setYearlyTab] = useState<TabInfo>({
+    name: 'Yearly',
+    content: [],
+  });
+  const [allTimeTab, setAllTimeTab] = useState<TabInfo>({
+    name: 'All Time',
+    content: [],
+  });
```

Comment:
Using React State in combination with Redux will get messy quickly. Related to my comment in the leaderboard component, we should pass the data directly into the component from the Redux store, which removes almost > 50 lines from this file.